src/sdk.js

```diff
@@ -114,9 +118,12 @@ export default class AccountsSDK {
   authorizeURL(options = {}, flow = '') {
     const localOptions = Object.assign({}, this.options, options);
 
+    if (!params.state) {
+      localOptions.state = random.string(localOptions.key_length);
+    }
+
     if (!localOptions.redirect_uri) {
-      localOptions.redirect_uri =
-        window.location.origin + window.location.pathname;
+      localOptions.redirect_uri = window.location.href;
```

Comment:
This change is no longer required. Href contains document fragments, we shouldn't use it here.
Let's use old redirect URIs and persist href in a different way.